src/CommunityStore/Shipping/Method/ShippingMethod.php

```diff
@@ -315,10 +338,26 @@ public static function getEligibleMethods()
             }
 
             if ($method->getShippingMethodTypeMethod()->isEligible()) {
+                $criteria = $method->getProductGroupsCriteria();
+                if ($criteria) {
+                    if ($criteriaService === null) {
+                        $criteriaService = app(Criteria::class);
+                    }
+                    if ($products === null) {
+                        $em = app(EntityManagerInterface::class);
+                        $products = [];
+                        foreach (Cart::getCart() as $entry) {
```

Comment:
This is the right way to get the items in the current cart, isn't it?

Reply:
Yes. It returns an array of items (so perhaps not the cleanest), but that would be the way to do it.